app/controllers/katello/api/v2/generic_content_units_controller.rb

```diff
@@ -0,0 +1,30 @@
+module Katello
+  class Api::V2::GenericContentUnitsController < Api::V2::ApiController
+    Katello::RepositoryTypeManager.generic_content_types(enabled_only: false).each do |type|
+      apipie_concern_subst(:a_resource => N_(type), :resource_id => type.pluralize)
+      resource_description do
+        name type.pluralize.titleize
+      end
+    end
+
+    include Katello::Concerns::Api::V2::RepositoryContentController
+
+    def default_sort
+      %w(name asc)
+    end
+
+    def resource_class
+      ::Katello::GenericContentUnit.where(content_type: params[:content_type].singularize)
+    end
+
+    def filter_by_content_type(collection)
+      collection.where(:content_type => params[:content_type])
```

Comment:
I don't think this method is needed anymore right? (with the resource_class method?)